tmcprototype/CentralNode/CentralNode/CentralNode.py

```diff
@@ -426,25 +439,72 @@ def AssignResources(self, argin):
                 }
                 }
 
-        :return: None.
+        Note: From Jive, enter input as:
+        {"subarrayID":1,"dish":{"receptorIDList":["0001"]}} without any space.
+
+        :return: The string in JSON format. The JSON contains following values:
+
+            dish:
+                Mandatory JSON object consisting of
+
+                receptorIDList_success:
+                    DevVarStringArray
+                    Contains ids of the receptors which are successfully allocated. Empty on unsuccessful allocation.
+
 
-        Note: From Jive, enter input as: {"subarrayID":1,"dish":{"receptorIDList":["0001"]}} without any space.
+            Example:
+                {
+                "dish": {
+                "receptorIDList_success": ["0001", "0002"]
+                        }
+                }
         """
+        receptorIDList = []
         try:
-            #serialize the json
+            # serialize the json
             jsonArgument = json.loads(argin)
-
-            #invoke command on subarray node
+            # Create subarray proxy
             subarrayID = int(jsonArgument['subarrayID'])
             subarrayProxy = self.subarray_FQDN_dict[subarrayID]
-            subarrayProxy.command_inout(CONST.CMD_ASSIGN_RESOURCES, jsonArgument["dish"]["receptorIDList"])
+            # Check for the duplicate receptor allocation
+            duplicate_allocation_count = 0
+            duplicate_allocation_dish_ids = []
+            for dish in range(0, len(jsonArgument["dish"]["receptorIDList"])):
```

Comment:
This is an inefficient way of loop iteration. For each iteration it will have to execute following steps:
1. navigate to ["dish"]["receptorIDList"] in the json object
2. calculate len
3. iterate inside the receptorIDList array till the index is reached.
This is ok if no of array elements are few. But for big array that will be a performance problem. I suggest following changes:
receptorIDList = jsonArgument["dish"]["receptorIDList"]
count =  len(receptorIDList)
for dish in range(0, count)):
    dish_ID = "dish" + receptorIDList[dish]
....

Reply:
Done. The commit id is: [5d54870ec6a8497900111874538350c0dc63f179](https://github.com/ska-telescope/tmc-prototype/commit/5d54870ec6a8497900111874538350c0dc63f179)